src/wolfgame.py

```diff
@@ -4126,73 +4102,62 @@ def check_exchange(cli, actor, nick):
-                else:
-                    var.START_VOTES.discard(nick)
-                    cli.msg(chan, messages["start_retract"].format(nick))
+            if not wrapper.source in var.START_VOTES:
+                wrapper.pm(messages["start_novote"])
+            else:
+                var.START_VOTES.discard(wrapper.source)
+                wrapper.send(messages["start_retract"].format(wrapper.source))
 
-                    if len(var.START_VOTES) < 1:
-                        var.TIMERS['start_votes'][0].cancel()
-                        del var.TIMERS['start_votes']
-            return
+                if len(var.START_VOTES) < 1:
+                    var.TIMERS["start_votes"][0].cancel()
+                    del var.TIMERS["start_votes"]
 
     if var.PHASE != "day":
         return
-    if nick in var.NO_LYNCH:
-        var.NO_LYNCH.remove(nick)
-        cli.msg(chan, messages["retracted_vote"].format(nick))
+    if wrapper.source in var.NO_LYNCH:
+        var.NO_LYNCH.remove(wrapper.source)
+        wrapper.send(messages["retracted_vote"].format(wrapper.source))
         var.LAST_VOTES = None # reset
         return
 
-    candidates = var.VOTES.keys()
-    for voter in list(candidates):
-        if nick in var.VOTES[voter]:
-            var.VOTES[voter].remove(nick)
+    for voter in list(var.VOTES):
+        if wrapper.source in var.VOTES[voter]:
+            var.VOTES[voter].remove(wrapper.source)
             if not var.VOTES[voter]:
                 del var.VOTES[voter]
-            cli.msg(chan, messages["retracted_vote"].format(nick))
+            wrapper.send(messages["retracted_vote"].format(wrapper.source))
             var.LAST_VOTES = None # reset
             break
     else:
-        cli.notice(nick, messages["pending_vote"])
+        wrapper.pm(messages["pending_vote"])
 
 @command("shoot", playing=True, silenced=True, phases=("day",))
 def shoot(var, wrapper, message):
     """Use this to fire off a bullet at someone in the day if you have bullets."""
-    if wrapper.target is not channels.Main:
-        return
-
-    if wrapper.source.nick not in var.GUNNERS.keys():
+    if wrapper.source not in var.GUNNERS.keys():
         wrapper.pm(messages["no_gun"])
         return
-    elif not var.GUNNERS.get(wrapper.source.nick):
+    elif not var.GUNNERS.get(wrapper.source):
         wrapper.pm(messages["no_bullets"])
         return
-    victim = get_victim(wrapper.source.client, wrapper.source.nick, re.split(" +", message)[0], True)
-    if not victim:
-        return
-    if victim == wrapper.source.nick:
-        wrapper.pm(messages["gunner_target_self"])
+
+    target = get_target(var, wrapper, re.split(" +", message)[0], not_self_message="gunner_target_self")
+    if not target:
         return
+
     # get actual victim
-    victim = choose_target(wrapper.source.nick, victim)
+    target = users._get(choose_target(wrapper.source.nick, target.nick)) # FIXME
```

Comment:
Use get_target() here and fire a targeted_command event (misdirection should apply for sure, unsure about exchange but that'd be funny too)